src/main/java/codeu/controller/ConversationServlet.java

```diff
@@ -32,110 +36,144 @@
+	request.setAttribute("groups", groups);
+    request.getRequestDispatcher("/WEB-INF/view/conversations.jsp").forward(request, response);
+  }
+
+  /**
+   * This function fires when a user submits the form on the conversations page. It gets the
+   * logged-in username from the session and the new conversation title from the submitted form
+   * data. It uses this to create a new Conversation object that it adds to the model.
+   */
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response)
+      throws IOException, ServletException {
+
+    String username = (String) request.getSession().getAttribute("user");
+    if (username == null) {
+      // user is not logged in, don't let them create a conversation
+      response.sendRedirect("/conversations");
+      return;
+    }
+
+    User user = userStore.getUser(username);
+
+    if (user == null) {
+      // user was not found, don't let them create a conversation
+      System.out.println("User not found: " + username);
+      response.sendRedirect("/conversations");
+      return;
+    }
+
+	String conversationTitle = "";
+	if(request.getParameter("conversationTitle") == null){
+		int counter = 0;
+		conversationTitle = (String) request.getSession().getAttribute("user") + "sGroup";
+		// while(conversationStore.isTitleTaken(conversationTitle)){
+		conversationTitle = (String) request.getSession().getAttribute("user") + "sGroup" + counter;
+		counter++;
+		// }
+	}else{
+		conversationTitle = request.getParameter("conversationTitle");
 	}
 
-	/**
-	 * Sets the UserStore used by this servlet. This function provides a common setup method for use
-	 * by the test framework or the servlet's init() function.
-	 */
-	void setUserStore(UserStore userStore) {
-		this.userStore = userStore;
-	}
-
-	/**
-	 * Sets the ConversationStore used by this servlet. This function provides a common setup method
-	 * for use by the test framework or the servlet's init() function.
-	 */
-	void setConversationStore(ConversationStore conversationStore) {
-		this.conversationStore = conversationStore;
-	}
-	
-	/*
-	 * Sets the ActivityStore used by this servlet. This function provides a common setup method
-	 * for use by the test framework of the servlet's init() function.
-	 */
-	 void setActivityStore(ActivityStore activityStore) {
-		this.activityStore = activityStore;
-	 }
-
-	/**
-	 * This function fires when a user navigates to the conversations page. It gets all of the
-	 * conversations from the model and forwards to conversations.jsp for rendering the list.
-	 */
-	@Override
-	public void doGet(HttpServletRequest request, HttpServletResponse response)
-			throws IOException, ServletException {
-		List<Conversation> conversations = conversationStore.getAllConversations();
-		request.setAttribute("conversations", conversations);
-		request.getRequestDispatcher("/WEB-INF/view/conversations.jsp").forward(request, response);
+    if (!conversationTitle.matches("[\\w*]*")) {
```

Comment:
So you also added restrictions to the conversation title ?

Reply:
I didn't change the restrictions, I just cut them and pasted them somewhere else in the document. I had to get rid of this check while I was building the functionality. 

The restrictions were already there in the original code that we were given

Reply:
Oh okay thanks for clarifying